jbang url to jar fails

running `jbang https://bintray.com/cardillo/maven/download_file?file_path=joinery%2Fjoinery-dataframe%2F1.9%2Fjoinery-dataframe-1.9-jar-with-dependencies.jar` fails with:

```
[jbang] Type in your choice (0 or 1) and hit enter. Times out after 10 seconds.
0
[jbang] Downloaded file https://d29vzk4ow07wi7.cloudfront.net/cd906b2f603b3979637eba885e208c195cb3a0a2013b6203840ddc9e6df31ad0?response-content-disposition=attachment%3Bfilename%3D%22joinery-dataframe-1.9-jar-with-dependencies.jar%22&Policy=eyJTdGF0ZW1lbnQiOiBbeyJSZXNvdXJjZSI6Imh0dHAqOi8vZDI5dnprNG93MDd3aTcuY2xvdWRmcm9udC5uZXQvY2Q5MDZiMmY2MDNiMzk3OTYzN2ViYTg4NWUyMDhjMTk1Y2IzYTBhMjAxM2I2MjAzODQwZGRjOWU2ZGYzMWFkMD9yZXNwb25zZS1jb250ZW50LWRpc3Bvc2l0aW9uPWF0dGFjaG1lbnQlM0JmaWxlbmFtZSUzRCUyMmpvaW5lcnktZGF0YWZyYW1lLTEuOS1qYXItd2l0aC1kZXBlbmRlbmNpZXMuamFyJTIyIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjA0ODY2MjcyfSwiSXBBZGRyZXNzIjp7IkFXUzpTb3VyY2VJcCI6IjAuMC4wLjAvMCJ9fX1dfQ__&Signature=jVUt1LLTh1L0Bwp53uqKkL1PpYCt~1RnEgPicWkwDqMDvkKN4jEffWVIE6CyrIliMOWXlD6X7XOab9gW4KYMFukaUMqPp1tkch3YsL5R81kpasMlnmpdvrqHb8Kz0XmOcNohIJj44EGvK1tjnT3WQ6mN2wZodtJR9V-Qm~3Roe7N~y7GWRm3UPr6~HqKbzaevsB19swddHSJW3JYKHza8x69zJwdEWSCEVb7MMzcuxs5AJE6T4uHaQoWGmN-Ko3eccUhrA-BXrOei19r7xRxR6iLK1jug-ksRZ45eDULGGDRL9VMmp2M1-WQa4nOEl3mwpNUzCag4VHHZ4QnMpaoQQ__&Key-Pair-Id=APKAIFKFWOMXM2UMTSFA
[jbang] System Java version detected as 11
[jbang] System Java version matches requested version 11
[jbang] run: /Users/max/.sdkman/candidates/java/11.0.8.hs-adpt/bin/java -classpath '/Users/max/code/personal/maxandersen/jbang-catalog/https:/bintray.com/cardillo/maven/download_file?file_path=joinery%2Fjoinery-dataframe%2F1.9%2Fjoinery-dataframe-1.9-jar-with-dependencies.jar' joinery.DataFrame
Error: Could not find or load main class joinery.DataFrame
Caused by: java.lang.ClassNotFoundException: joinery.DataFrame
```

download happens but filename fails.

--- a/src/main/java/dev/jbang/ArtifactInfo.java
+++ b/src/main/java/dev/jbang/ArtifactInfo.java
@@ -28,7 +28,7 @@ public class ArtifactInfo {
 
 	public String toString() {
 		String path = asFile().getAbsolutePath();
-		return getCoordinate().toCanonicalForm() + "=" + path;
+		return getCoordinate() == null ? "<null>" : getCoordinate().toCanonicalForm() + "=" + path;
 	}
 
 	@Override

--- a/src/main/java/dev/jbang/Script.java
+++ b/src/main/java/dev/jbang/Script.java
@@ -265,7 +265,11 @@ public class Script {
 					classpath = new DependencyUtil().resolveDependencies(additionalDeps,
 							Collections.emptyList(), offline, !Util.isQuiet());
 				} else {
-					classpath = new ModularClassPath(Arrays.asList(new ArtifactInfo(null, new File(originalFile))));
+					if (getBackingFile() == null) {
+						classpath = new ModularClassPath(Arrays.asList(new ArtifactInfo(null, new File(originalFile))));
+					} else {
+						classpath = new ModularClassPath(Arrays.asList(new ArtifactInfo(null, getBackingFile())));
+					}
 				}
 				// fetch main class as we can't use -jar to run as it ignores classpath.
 				if (getMainClass() == null) {

--- a/src/test/java/dev/jbang/cli/TestRun.java
+++ b/src/test/java/dev/jbang/cli/TestRun.java
@@ -149,6 +149,31 @@ public class TestRun {
 	}
 
 	@Test
+	void testJarViaHttps(@TempDir Path tdir) throws IOException {
+
+		String jar = "https://bintray.com/cardillo/maven/download_file?file_path=joinery%2Fjoinery-dataframe%2F1.9%2Fjoinery-dataframe-1.9-jar-with-dependencies.jar";
+
+		try {
+			Settings.getTrustedSources().add(jar, tdir.resolve("test.trust").toFile());
+
+			environmentVariables.clear("JAVA_HOME");
+			Jbang jbang = new Jbang();
+
+			CommandLine.ParseResult pr = new CommandLine(jbang).parseArgs("run", jar);
+			Run run = (Run) pr.subcommand().commandSpec().userObject();
+
+			Script result = prepareScript(jar, run.userParams, run.properties, run.dependencies, run.classpaths);
+
+			String cmdline = run.generateCommandLine(result);
+
+			assertThat(cmdline, not(containsString("https")));
+
+		} finally {
+			Settings.getTrustedSources().remove(Arrays.asList(jar), tdir.resolve("test.trust").toFile());
+		}
+	}
+
+	@Test
 	void testHelloWorldGAVWithNoMain() throws IOException {
 
 		environmentVariables.clear("JAVA_HOME");

